department-of-veterans-affairs/vets-api
# Champva 94284 pega reporting api connection

## Summary

- This work is behind a feature toggle (flipper): NO
- Created a client for the Pega API with some tests to make use of it.
- We will need to retrieve data from this API in upcoming tickets.  With this ticket we have implemented a simple API client and some test code as a proof of concept.
- I work for the IVC Forms team and we own this module.

## Related issue(s)

- [The associated ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94284)

## Testing done

- Prior to this change there was no client for the Pega API.  With this change, there is a client that is only used by unit test code.
- In addition to running the test suite, verification can be performed by:
  - Enable the 'hit the production endpoint' test
  - Configure the API key in your local environment as described in the comments
  - Run this test locally
  - Verify the response using a debugger

## Screenshots
None

## What areas of the site does it impact?
None, this code is only used by unit tests so far.

## Acceptance criteria


## Requested Feedback
None
